packages/web3-utils/src/converters.ts

```diff
@@ -555,17 +558,30 @@ export const toWei = (number: Numbers, unit: EtherUnits): string => {
 	if (!denomination) {
 		throw new InvalidUnitError(unit);
 	}
+	let parsedNumber = number;
+	if (typeof parsedNumber === 'number'){
+		if (parsedNumber  < 1e-15){
+			console.warn(PrecisionLossWarning)
+		}
+		if (parsedNumber > 1e+20) {
+			console.warn(PrecisionLossWarning)
 
+			parsedNumber =  BigInt(parsedNumber);
+		}
+	}
+	
+	parsedNumber = typeof number === 'number' ? number.toLocaleString('fullwide', {useGrouping: false, maximumFractionDigits: 20}) : number;
```

Comment:
I think it is more clear to put the variable assigmnet here and then no need to use `? :`
```suggestion
		parsedNumber = number.toLocaleString('fullwide', {useGrouping: false, maximumFractionDigits: 20});
	}

```

Reply:
Apologies, this should be 
`parsedNumber = typeof parsedNumber === 'number' ? parsedNumber.toLocaleString('fullwide', {useGrouping: false, maximumFractionDigits: 20}) : parsedNumber;`

We only want to use `toLocaleString`on type `number` as there is a limit on fraction digits (20 in this case, javascript only supports up to 17 decimal places). If this was used on BigInt and String the input may lose precision on decimal places